tools/symbols_map_generator/main.py

```diff
@@ -147,8 +147,23 @@ def get_namespace_str(node: clang.cindex.Cursor) -> list[str]:
     
     return get_namespace_str(node.semantic_parent) + [spelling]
 
+def symbol_passes_quirk_check(library_name: str, symbol: str):
+    """
+    Checks if the provided 
+    """
+
+    # WaylandExtensions::Context ctor and dtor symbols are inline
+    # but can be exported by unoptimized builds.
+    # This makes the "regenerate" script hide them:
+    if library_name == "miral":
+        return (symbol != "miral::SessionLockListener::?SessionLockListener*;"
+            and symbol != "miral::WaylandExtensions::Context::?Context*;"
+            and symbol != "miral::WaylandExtensions::Context::Context*;"
+            and symbol != "vtable?for?miral::WaylandExtensions::Context;")
```

Comment:
Couldn't this be a constant `set()` at the top, and then this would be a simple `symbol in HIDDEN_SYMBOLS` or such?